http-client/src/main/java/io/airlift/http/client/HttpClientConfig.java

```diff
@@ -74,6 +75,14 @@ public class HttpClientConfig
     private DataSize maxContentLength = DataSize.of(16, MEGABYTE);
     private DataSize requestBufferSize = DataSize.of(4, KILOBYTE);
     private DataSize responseBufferSize = DataSize.of(16, KILOBYTE);
+    private DataSize maxHeapMemory = clamp(
+            DataSize.ofBytes(Runtime.getRuntime().maxMemory() / 8),
+            DataSize.of(128, MEGABYTE),
```

Comment:
What's the minimum value that will possibly work. E.g., if you run an application with a heap of 100 MB, what does jetty do by default?

Reply:
it will assign 1/8th of the memory

Reply:
but 100 mb isn't enough for the airlift based application to even load I think

Reply:
I can remove the @MaxSize but I think that it should be clamped either way. 32 GB for the http-server, even a busy one, is enough

Reply:
I've dropped @MaxDataSize and lowered @MinDataSize and default value to 16 MB. This is extremely low value that is highly unlikely to occur in any real-world deployment scenario as this would imply 128 MB heap which is not enough to start most Java applications.

Reply:
>  which is not enough to start most Java applications.

It depends on what application you're talking about. Simple well-written clients can run with limited memory. The Trino CLI can run with as little as 30MB, for instance.

Reply:
trino-cli isn't using jetty client